misc/scripts/plugin_sanity_test.sh

```diff
@@ -0,0 +1,122 @@
+#!/bin/bash
+
+# Copyright 2017 VMware, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# To sanity test on the same VM using managed plugin
+# More information https://github.com/vmware/docker-volume-vsphere/issues/1020#issue-213470358
+
+echo "plugin_sanity_test: [INFO] Running plugin_sanity_test on the clean test setup..."
+
+# make sure sanity check is on clean slate and row count should be 2
+row_count=`docker plugin ls | wc -l`
+if [ $row_count != 2 ]; then
+    echo "plugin_sanity_test: [ERROR] error has occurred while counting pluginName row...current count is " $row_count
+    exit 1
+fi
+
+# get installed plugin name
+pluginName=`docker plugin ls | sed 1,1d | awk '{ print $2 }'`
+echo "plugin_sanity_test: [INFO] Installed plugin name is:" $pluginName
+
+# make sure plugin name is not empty
+if [ -z $pluginName ]; then
+    echo "plugin_sanity_test: [ERROR] error has occurred while fetching created plugin name..
+         please make sure plugin is built correctly or not"
+         exit 1
+fi
+
+# check plugin is disabled or not .. by default it should be disabled
+echo "plugin_sanity_test: [INFO] check plugin is disabled or not .. by default it should be disabled..."
+enabled=`docker plugin inspect $pluginName -f {{.Enabled}}`
+if [ $? -ne 0  -o  $enabled != "false" ]; then
+        echo "[ERROR] plugin_sanity_test: docker plugin inspect failed..[" $enabled "]"
+        exit 1
+fi
+
+# setting random value for plugin name
+volumeName="volume_`date +%Y%m%d%H%M%S`";
+
+# create volume when plugin is disabled state and expects an error
+echo "plugin_sanity_test: [INFO] create volume when plugin is disabled state and expects an error..."
+docker volume create -d $pluginName --name $volumeName
+if [ $? -eq 0 ]; then
+        echo "[ERROR] plugin_sanity_test: volume should not be created as plugin is not enabled yet"
+        exit 1
+fi
+
+# Enable plugin and verifies it is enabled or not
+echo "plugin_sanity_test: [INFO] Enable plugin and verifies it is enabled or not..."
+docker plugin enable $pluginName
```

Comment:
same here. I understand the test. but it seems to be more for docker functionality. I think we need to minimize cross-testing docker when it is not expected to impact our code